controllers/idle_handlers.go

```diff
@@ -110,11 +83,128 @@ func (r *ImageBasedUpgradeReconciler) handleFinalizeFailure(ctx context.Context,
 	return doNotRequeue(), nil
 }
 
+// cleanup cleans stateroots, precache, backup, ibu files and OADP operator
+// returns true if all cleanup tasks were successful
+func (r *ImageBasedUpgradeReconciler) cleanup(
+	ctx context.Context, allUnbootedStateroots bool,
+	ibu *lcav1alpha1.ImageBasedUpgrade) (bool, string) {
+	// try to clean up as much as possible and avoid returning when one of the cleanup tasks fails
+	// successful means that all the cleanup tasks completed without any error
+	successful := true
+	errorMessage := ""
+
+	var handleError = func(err error, msg string) {
+		successful = false
+		r.Log.Error(err, msg)
+		errorMessage += msg + " "
+	}
+
+	if err := r.cleanupStateroots(allUnbootedStateroots, ibu); err != nil {
+		handleError(err, "failed to cleanup stateroots.")
```

Comment:
nit: This would break the linter rules for a fmt.Errorf, which does not allow the message to end in punctuation. We should try to follow the same rules for consistency.

I guess the intent here is that we're running all steps, regardless of failure, and potentially building a single error message with multiple components, so the period is intended to separate those?

Reply:
yes